src/textual/css/scalar_animation.py

```diff
@@ -48,11 +51,18 @@ def __init__(
             assert duration is not None, "Duration expected to be non-None"
             self.duration = duration
 
-    def __call__(self, time: float) -> bool:
+    def __call__(
+        self, time: float, app_animation_level: AnimationLevel = "full"
+    ) -> bool:
         factor = min(1.0, (time - self.start_time) / self.duration)
         eased_factor = self.easing(factor)
 
-        if eased_factor >= 1:
+        if (
```

Comment:
I guess this sets the value to its final position, but what is to prevent the animator calling it multiple times? Anything?

Reply:
Yeah. The `return True` tells the animator the animation was completed, which means that the animation will have its "on complete callback" called (if there's any) and then the animator gets rid of the animation:

https://github.com/Textualize/textual/blob/16656e62e47943e65d06bdb1737decdaf4ad59f3/src/textual/_animator.py#L528-L532